### PR TITLE
Make figure reporting configureable, and figures quick navigable

### LINKED ISSUE #10826: Add script for aria figure to input gestures and give the user the possibility to turn off reporing of this role in document formating settings
### Is your feature request related to a problem? Please describe.
See examples for Aria figure here
https://www.w3.org/WAI/tutorials/images/groups/#multiple-images-conveying-a-single-piece-of-information

I see more and more websites using this aria role to group information. The problem is that NVDA alway announce when entering and leaving this container. While this might be useful in some circumstances, users will most probably prefer to decide themselves if this should be reported or not while navigating in browse mode.

### Describe the solution you'd like
The best solution is to maintain consistency here. I.e. as it is in case of groupings, there should be a script for figures in input gestures. And the users should be able to turn off reporting in document formating settings.

### Describe alternatives you've considered
None

### Additional context
None

### LINKED ISSUE #14349: Add option to configure figure and caption reporting
### Is your feature request related to a problem? Please describe.
Currently, there is no way to configure reporting of figure or caption elements when using browse mode.

### Describe the solution you'd like
Either have one checkbox that would toggle both figures and captions together, or a separate one for each. 

### Additional context
Sample [source view](https://codepen.io/reefturner/pen/YzvEWYw)

- Using Firefox visit: [Full view](https://codepen.io/reefturner/full/YzvEWYw)
- Move the virtual cursor to "button before" in browse mode.
- Press down arrow:

```
NVDA+space
Browse mode
NVDA+up arrow
button    before
down arrow
figure    caption    Media caption XYZ
```

### PR DESCRIPTION
**Pull request by SaschaCowley**
### Link to issue number:

fixes #10826 
Fixes #14349 

### Summary of the issue:

It has hitherto not been possible to navigate by figure with quick navigation keys, or to disable the reporting of figures and captions.

### Description of user facing changes

Added a script to jump to next/previous figure in browse mode. Added an option to disable the reporting of figures and captions, and a script to toggle this setting.

### Description of development approach

Examined similar code and based new code on existing code.

### Testing strategy:

Manually tested the changes using a [demo page](https://saschacowley.com/demos/figure.html). Ran unit and translation tests.

### Known issues with pull request:

Only tested to work in firefox, as there is a problem with figures in Chromium (#16137).

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [ ] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

### CODE REVIEW

**Comment by cary-rowen** [comment]
Thanks for your efforts, this is great. I looked at the source code and noticed you assigned the y key to this, could you mention this gesture in the description?

**Comment by SaschaCowley** [comment]
> I looked at the source code and noticed you assigned the y key to this, could you mention this gesture in the description?

Bother, I meant to remove this assignment before opening the PR. I shall remove it now, as I don't think most users will want this assigned by default.

File: user_docs/en/changes.t2t [hunk 1]
```diff
@@ -8,8 +8,9 @@ What's New in NVDA
 == New Features ==
 - New key commands:
   - New Quick Navigation command ``p`` for jumping to next/previous text paragraph in browse mode. (#15998, @mltony)
-  -
+  - New unassigned quick navigation commands to jump to the next/previous figure. (#10826)
```
Comment (seanbudd) [inline user_docs/en/changes.t2t:11]: t2t lists need a list terminator

```suggestion
  - New unassigned Quick Navigation command to jump to the next/previous figure. (#10826)
  -
```

File: user_docs/en/changes.t2t [hunk 2]
```diff
@@ -19,7 +20,7 @@ What's New in NVDA
       - Laptop: ``NVDA+control+shift+pageUp`` or ``NVDA+control+shift+pageDown``.
       -
     -
-  -
+  - Added a new input gesture to toggle the reporting of figures and captions. (#10826, #14349)
```
Comment (seanbudd) [inline user_docs/en/changes.t2t:23]:
```suggestion
  - Added a new unassigned input gesture to toggle the reporting of figures and captions. (#10826, #14349)
  -
```

**Review by seanbudd — APPROVED** [review]

**Review by Qchristensen — APPROVED** [review]
Looks good, great job Sascha!

**Comment by SaschaCowley** [comment]
Disabling reporting of figures and captions as implemented here also disables reporting of table captions. While usage is quite similar, I feel that users may want to be able to configure table caption reporting in table formatting settings. Thoughts, @Adriani90, @michaelDCurran?

**Comment by Adriani90** [comment]
There is issue #14874 which mostly leads to the reason in general why users would disable reporting of captions. This also affects table captions.
I agree though table captions should be treated separately in the table formating settings.
For captions which span on multiple lines, In my view hearing the "title" from captions when navigating to them might be useful, but if I navigate the caption with arrow keys then i expect NVDA to report "title" only when entering the caption and "end of title" when leaving the caption.
However, this affects also other elements displayed on multiple lines such as links, buttons etc.

**Pull request merged**